packages/cloud_firestore/cloud_firestore_platform_interface/lib/cloud_firestore_platform_interface.dart

```diff
@@ -0,0 +1,163 @@
+library cloud_firestore_platform_interface;
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+import 'dart:ui';
+import 'dart:math';
+
+import 'package:collection/collection.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show required, visibleForTesting;
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+part 'src/method_channel_firestore.dart';
+part 'src/blob.dart';
+part 'src/utils/auto_id_generator.dart';
+part 'src/method_channel_field_value_factory.dart';
+part 'src/platform_interface/field_value_factory.dart';
+part 'src/platform_interface/collection_reference.dart';
+part 'src/method_channel_collection_reference.dart';
+part 'src/method_channel_document_change.dart';
+part 'src/platform_interface/document_change.dart';
+part 'src/method_channel_document_reference.dart';
+part 'src/platform_interface/document_reference_interface.dart';
+part 'src/document_snapshot.dart';
+part 'src/field_path.dart';
+part 'src/platform_interface/field_value.dart';
+part 'src/firestore_message_codec.dart';
+part 'src/geo_point.dart';
+part 'src/method_channel_query.dart';
+part 'src/platform_interface/query.dart';
+part 'src/platform_interface/query_snapshot.dart';
+part 'src/method_channel_query_snapshot.dart';
+part 'src/snapshot_metadata.dart';
+part 'src/source.dart';
+part 'src/timestamp.dart';
+part 'src/transaction.dart';
+part 'src/transaction_platform_interface.dart';
+part 'src/write_batch.dart';
+part 'src/write_batch_platform_interface.dart';
+
+/// Defines an interface to work with [FirestorePlatform] on web and mobile
+abstract class FirestorePlatform extends PlatformInterface {
+  /// [FirebaseApp] used for this firestore instance
+  final FirebaseApp app;
+
+  /// Create an instance using [app]
+  FirestorePlatform({FirebaseApp app})
+      : app = app ?? FirebaseApp.instance,
+        super(token: _token);
+
+  static final Object _token = Object();
+
+  /// Create an instance using [app] using the existing implementation
+  factory FirestorePlatform.withApp({FirebaseApp app}) {
```

Comment:
I found it confusing that you can also create a `FirestorePlatform` using the default constructor above.